### ISSUE TITLE
convert docs to roxygen2 ?

### ISSUE THREAD

**Issue by dankelley** [dankelley/ocedata#17]
I noticed that the docs are hand-written.  I think it would be smart to transition to roxygen2, as I did for `oce` a while back.  Otherwise, a new contributor would have to know Rd format, or at least learn enough of it to add material.  And, frankly, I don't see a lot of merit in using Rd when Roxygen2 is perhaps better-known, and also simpler and less error-prone (less time needs to be spent on matching `{`, etc).

I see 27 Rd files, totalling 1064 lines.  With decent vim macros, I can do a lot of the conversion fairly easily, e.g. transforming **\code{blah}** to **\`blah\`** (it was harder to figure out to typeset that for GH than it is to write the vim macro).

If I do this, it will be in a separate branch that would only be merged after I used it a while.

**Comment by dankelley** [comment]
The new branch is called `roxygen` and it hangs off `develop`, and will be merged there, and thereafter to `main`, if things go well.

I don't plan to start this until the weekend.

**Comment by dankelley** [comment]
I am going to reject this suggestion and close the issue.  I've tried a simple case, removing all files in `man/` and then creating a simple `soi.R` (see Details) but I cannot build the package, instead receiving an error (again see Details).  We have something tricky here, where one of the `data` items is not an rda, but rather an `R` file, and maybe that has something to do with it.

I've done some web searching to get clues, but to no avail.  Assessing that the result cannot justify the effort, I will up on this idea.  I didn't get far into the work, so I'll remove the `roxygen` branch.

<details>

**R/soi.R**

```R
#' Southern Oscillation Index
#'
#' `soi` is a data frame containing the Southern Oscillation Index with
#' columns named `year` for decimal year (sampled monthly and represented
#' at mid-month) and `index` for the index.
#'
#' @name soi
#'
#' @docType data
#'
#' @usage data(soi)
NULL
```

**Error**

```
==> devtools::document(roclets = c('rd'))

ℹ Updating ocedata documentation
First time using roxygen2. Upgrading automatically...
ℹ Loading ocedata
Error in readChar(con, 5L, useBytes = TRUE) : cannot open the connection
Calls: suppressPackageStartupMessages ... eval.parent -> eval -> eval -> eval -> eval -> load -> readChar
In addition: Warning message:
In readChar(con, 5L, useBytes = TRUE) :
  cannot open compressed file '', probable reason 'No such file or directory'
Execution halted

Exited with status 1.
```

**Comment by dankelley** [comment]
I figured out why it wasn't building, and reinstated the "roxygen2" branch. I've done pre-release things (spelling check, URL check, ...) and am fixing up problems I'm finding.  As is expected for something not updated in a couple of years, some URLs need alteration and some are just plain nonexistent now.

I'm glad I came back to this, because roxygen is a lot easier to maintain.  (I doubt that many new developers are learning Rd at all, anymore ... what would be the point?)
